openfecwebapp/views.py

```diff
@@ -185,35 +185,61 @@ def groupby(values, keygetter):
 def render_candidate(candidate, committees, cycle, election_full=True):
     # candidate fields will be top-level in the template
     tmpl_vars = candidate
-
     tmpl_vars['parent'] = 'data'
     tmpl_vars['cycle'] = cycle
     tmpl_vars['result_type'] = 'candidates'
     tmpl_vars['duration'] = election_durations.get(candidate['office'], 2)
     tmpl_vars['election_full'] = election_full
+    tmpl_vars['report_type'] = report_types.get(candidate['office'])
+    tmpl_vars['context_vars'] = {'cycles': candidate['cycles'], 'name': candidate['name']}
+
+    # In the case of when a presidential or senate candidate has filed
+    # for a future year that's beyond the current cycle,
+    # set a max_cycle var to the current cycle we're in
+    # and when calling the API for totals, set election_full to False.
+    # The max_cycle value is also referenced in the templates for setting
+    # the cycle for itemized tables. Because these are only in 2-year chunks,
+    # the cycle should never be beyond the one we're in.
+    tmpl_vars['cycles'] = [cycle for cycle in candidate['cycles'] if cycle <= utils.current_cycle()]
+    tmpl_vars['max_cycle'] = cycle if cycle <= utils.current_cycle() else utils.current_cycle()
+    tmpl_vars['show_full_election'] = election_full if cycle <= utils.current_cycle() else False
+
+    # Annotate committees with most recent available cycle
     tmpl_vars['aggregate_cycles'] = (
         list(range(cycle, cycle - tmpl_vars['duration'], -2))
         if election_full
         else [cycle]
     )
-
-    # Annotate committees with most recent available cycle
     for committee in committees:
         committee['related_cycle'] = (
             max(cycle for cycle in tmpl_vars['aggregate_cycles'] if cycle in committee['cycles'])
             if election_full
             else candidate['two_year_period']
         )
 
+    # Group the committees by designation
     committee_groups = groupby(committees, lambda each: each['designation'])
     committees_authorized = committee_groups.get('P', []) + committee_groups.get('A', [])
 
+    tmpl_vars['committee_groups'] = committee_groups
```

Comment:
I did some general moving code around and adding comments, so some of these are false diffs.